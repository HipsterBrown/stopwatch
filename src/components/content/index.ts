export { default as PageContent } from './content';
