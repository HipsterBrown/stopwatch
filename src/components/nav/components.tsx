import styled from '../../styled';

export const Menu = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  z-index: 99;
  display: inline-block;
  width: 70px;
  height: 100vh;
  transition: transform 0.6s, opacity 0.6s, -webkit-transform 0.6s;
  will-change: transform, opacity;
`;

export const MenuPrimary = styled.div`
  position: relative;
  display: inline-block;
  width: 70px;
  height: 100%;
  vertical-align: top;
`;

export const MenuBackground = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: ${({ theme }) => theme.color.navbar};
  transition: transform 0.6s;
  transition: transform 0.6s, -webkit-transform 0.6s;
  transform-origin: left center;
  will-change: transform;
`;

export const MenuContainer = styled.div`
  position: relative;
  height: 100%;
  z-index: 1;
`;

export const LogoContainer = styled.span`
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -25px;
  margin-left: -25px;
  transition: background 0.2s, transform 0.2s;
  pointer-events: none;
  color: #fff;
  justify-content: center;
  align-items: center;
  display: flex;
  width: 50px;
  height: 50px;
  line-height: 34px;
  padding: 8px;
  z-index: 999;
  border-radius: 12px;
  background: rgba(255, 255, 255, 0.1);
  box-shadow: 0 8px 26px 0 rgba(43, 80, 237, 0.31);
`;

export const LogoWrapper = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  display: inline-block;
  height: 100px;
  white-space: nowrap;
  &:hover {
    & > section {
      opacity: 1;
      transform: translate3d(0, 0, 0);
      pointer-events: auto;
    }

    & > span {
      transform: translate3d(10px, 0, 0);
      background: ${({ theme }) => theme.color.navbar};
    }
  }
`;

export const LogoArrow = styled.i`
  position: absolute;
  bottom: 0;
  left: 50%;
  color: rgba(207, 215, 227, 0.6);
  font-size: 16px;
  transform: translate3d(-50%, 0, 0);
`;

export const TabWrapper = styled.div`
  position: absolute;
  top: 50%;
  left: 0;
  transform: translate3d(0, -50%, 0);
`;
