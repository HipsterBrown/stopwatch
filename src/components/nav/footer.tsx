import React from 'react';
import { Sun, Minimize } from 'react-feather';

import styled from '../../styled';

export const FooterWrapper = styled.div`
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
`;

export const FooterItemWrapper = styled.span`
  position: relative;
  display: inline-block;
  line-height: 70px;
  height: 100%;
  white-space: nowrap;
`;

export const FooterItem = styled.span`
  display: inline-block;
  width: 40px;
  height: 40px;
  line-height: 40px;
  margin: 0 15px;
  vertical-align: middle;
  color: #fff;
  text-align: center;
  border-radius: 12px;
  cursor: pointer;
  transition: transform 0.2s, opacity 0.2s, color 0.2s, -webkit-transform 0.2s;
  &:hover {
    color: rgba(255, 255, 255, 0.6);
  }
`;

export const FooterItemIcon = styled.span`
  position: relative;
  display: inline-block;
  color: #fff;
  width: 32px;
  height: 32px;
  line-height: 32px;
  font-size: 20px;
  text-align: center;
  vertical-align: middle;
  z-index: 1;
  cursor: pointer;
  transition: transform 0.2s, opacity 0.2s, color 0.2s, -webkit-transform 0.2s;
  &:hover {
    color: rgba(255, 255, 255, 0.6);
  }

  & > svg {
    cursor: pointer;
  }
`;

function ThemeSwitcher() {
  return (
    <FooterItemWrapper>
      <FooterItem>
        <FooterItemIcon>
          <Sun strokeWidth={1.5} size={24} />
        </FooterItemIcon>
      </FooterItem>
    </FooterItemWrapper>
  );
}

function WindowSwitcher() {
  return (
    <FooterItemWrapper>
      <FooterItem>
        <FooterItemIcon>
          <Minimize strokeWidth={1.5} size={24} />
        </FooterItemIcon>
      </FooterItem>
    </FooterItemWrapper>
  );
}

export default function NavFooter() {
  return (
    <FooterWrapper>
      <WindowSwitcher />
      <ThemeSwitcher />
    </FooterWrapper>
  );
}
