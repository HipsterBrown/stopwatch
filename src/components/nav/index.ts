export { default as Nav } from './nav';
export { default as LogoDropdown } from './logo-dropdown';
export { default as NavFooter } from './footer';
