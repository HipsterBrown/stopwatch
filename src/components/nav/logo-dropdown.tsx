import React from 'react';
import { Info, HelpCircle, Meh, LogOut } from 'react-feather';

import styled from '../../styled';

export const Dropdown = styled.section`
  position: absolute;
  top: 0;
  left: 0;
  width: 300px;
  background: #fff;
  box-shadow: 0 20px 60px 0 rgba(0, 0, 0, 0.15);
  border-radius: 0 0 12px;
  z-index: 99;
  opacity: 0;
  overflow: auto;
  transform: translate3d(-60px, 0, 0);
  transition: transform 0.2s, opacity 0.2s, -webkit-transform 0.2s;
  pointer-events: none;
`;

export const SectionWrapper = styled.div`
  padding: 10px 0;
  border-bottom: 1px solid #f3f5f8;
  justify-content: flex-end;
  display: flex;
  white-space: nowrap;

  &:last-of-type {
    border-bottom: none;
  }
`;

export const Title = styled(SectionWrapper)`
  padding: 10px 0;
  border-bottom: 1px solid #f3f5f8;

  & h2 {
    font-family: ${({ theme }) => theme.font.regular};
    font-size: 25px;
    color: #000;
    margin: 0;
    padding: 0;
    margin-bottom: 5px;
  }

  & span {
    font-size: 11px;
    color: #b6bfcc;
    font-family: ${({ theme }) => theme.font.regular};
  }
`;

export const TitleSection = styled.div`
  padding: 15px 15px 10px;
  width: 73%;
`;

export const SectionItem = styled.span`
  width: 100%;
`;

export const SectionItemTitle = styled.p`
  color: #b6bfcc;
  font-size: 10px;
  font-family: ${({ theme }) => theme.font.medium};
  text-transform: uppercase;
  padding: 0 20px;
  margin-bottom: 8px;
`;

export const MenuItem = styled.a`
  display: block;
  height: 40px;
  padding: 0 20px;
  line-height: 40px;
  font-size: 12px;
  color: #768191;
  font-family: ${({ theme }) => theme.font.medium};
  text-align: left;
  white-space: nowrap;
  transition: color 0.2s;
  text-decoration: none;
  cursor: pointer;
  &:hover,
  &:visited {
    color: #000;
    text-decoration: none;
  }

  span {
    display: inline-block;
    width: 22px;
    margin-right: 12px;
    font-size: 18px;
    text-align: center;
    vertical-align: middle;
    opacity: 0.7;
  }
`;

export default function LogoDropdown() {
  return (
    <Dropdown>
      <Title>
        <TitleSection>
          <h2>Stopwatch</h2>
          <span>Track time like a thing</span>
        </TitleSection>
      </Title>
      <SectionWrapper>
        <SectionItem>
          <SectionItemTitle>Menu</SectionItemTitle>
          <MenuItem href="#">
            <span>
              <Info size={18} />
            </span>
            About Stopwatch
          </MenuItem>
          <MenuItem href="#">
            <span>
              <HelpCircle size={18} />
            </span>
            Help
          </MenuItem>
          <MenuItem href="#">
            <span>
              <Meh size={18} />
            </span>
            Report Issue
          </MenuItem>
          <MenuItem href="#">
            <span>
              <LogOut size={18} />
            </span>
            Quit Window
          </MenuItem>
        </SectionItem>
      </SectionWrapper>
    </Dropdown>
  );
}
