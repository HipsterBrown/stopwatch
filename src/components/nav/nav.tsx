import React, { ReactNode } from 'react';
import { ChevronDown } from 'react-feather';
import {
  Menu,
  MenuPrimary,
  MenuBackground,
  MenuContainer,
  LogoWrapper,
  TabWrapper,
  LogoArrow,
  LogoContainer,
} from './components';
import LogoDropdown from './logo-dropdown';
import { ReactComponent as Logo } from '../../assets/logo.svg';
import NavFooter from './footer';

export default function Nav(props: { children: ReactNode }) {
  return (
    <Menu>
      <MenuPrimary>
        <MenuBackground />
        <MenuContainer>
          <LogoWrapper>
            <LogoDropdown />
            <LogoContainer>
              <Logo />
            </LogoContainer>

            <LogoArrow>
              <ChevronDown strokeWidth={1} size={12} />
            </LogoArrow>
          </LogoWrapper>
          <TabWrapper>{props.children}</TabWrapper>
          <NavFooter />
        </MenuContainer>
      </MenuPrimary>
    </Menu>
  );
}
