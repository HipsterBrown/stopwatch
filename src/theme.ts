import { Theme } from './styled';

const theme: Theme = {
  color: {
    navbar: '#2B50ED',
    background: '#f8fafc',
  },
  font: {
    regular: 'Graphik-Regular',
    bold: 'Graphik-Bold',
    super: 'Graphik Super',
    medium: 'Graphik-Medium',
    semi: 'Graphik-Semibold',
    light: 'Graphik-Light',
  },
};

export default theme;
