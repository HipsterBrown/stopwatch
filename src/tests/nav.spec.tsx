import renderer from 'react-test-renderer';
import React, { ReactNode } from 'react';
import { ThemeProvider } from 'emotion-theming';
import theme from '../theme';
import { Nav, NavFooter, LogoDropdown } from '../components/nav';

describe('<Nav />', () => {
  let ThemeWrapper;
  beforeEach(() => {
    ThemeWrapper = (props: { children: ReactNode }) => (
      <ThemeProvider theme={theme}>{props.children}</ThemeProvider>
    );
  });
  it('renders without correctly', () => {
    expect(
      renderer
        .create(
          <ThemeWrapper>
            <Nav />
          </ThemeWrapper>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });

  it('renders footer correctly', () => {
    expect(
      renderer
        .create(
          <ThemeWrapper>
            <NavFooter />
          </ThemeWrapper>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });

  it('renders logo dropdown correctly', () => {
    expect(
      renderer
        .create(
          <ThemeWrapper>
            <LogoDropdown />
          </ThemeWrapper>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
