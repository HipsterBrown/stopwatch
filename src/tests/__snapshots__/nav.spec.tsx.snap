// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders footer correctly 1`] = `
<div
  className="css-16jk3lx"
>
  <span
    className="css-10abcsd"
  >
    <span
      className="css-7r5427"
    >
      <span
        className="css-1mgqi2s"
      >
        <svg
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={1.5}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 3v3a2 2 0 0 1-2 2H3m18 0h-3a2 2 0 0 1-2-2V3m0 18v-3a2 2 0 0 1 2-2h3M3 16h3a2 2 0 0 1 2 2v3"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="css-10abcsd"
  >
    <span
      className="css-7r5427"
    >
      <span
        className="css-1mgqi2s"
      >
        <svg
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={1.5}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="5"
          />
          <line
            x1="12"
            x2="12"
            y1="1"
            y2="3"
          />
          <line
            x1="12"
            x2="12"
            y1="21"
            y2="23"
          />
          <line
            x1="4.22"
            x2="5.64"
            y1="4.22"
            y2="5.64"
          />
          <line
            x1="18.36"
            x2="19.78"
            y1="18.36"
            y2="19.78"
          />
          <line
            x1="1"
            x2="3"
            y1="12"
            y2="12"
          />
          <line
            x1="21"
            x2="23"
            y1="12"
            y2="12"
          />
          <line
            x1="4.22"
            x2="5.64"
            y1="19.78"
            y2="18.36"
          />
          <line
            x1="18.36"
            x2="19.78"
            y1="5.64"
            y2="4.22"
          />
        </svg>
      </span>
    </span>
  </span>
</div>
`;

exports[`<Nav /> renders logo dropdown correctly 1`] = `
<section
  className="css-1vco01a"
>
  <div
    className="css-k9ylf6"
  >
    <div
      className="css-pfryns"
    >
      <h2>
        Stopwatch
      </h2>
      <span>
        Track time like a thing
      </span>
    </div>
  </div>
  <div
    className="css-1koizjm"
  >
    <span
      className="css-1ago99h"
    >
      <p
        className="css-1pbk3k6"
      >
        Menu
      </p>
      <a
        className="css-1dtgp6n"
        href="#"
      >
        <span>
          <svg
            fill="none"
            height={18}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="8"
            />
          </svg>
        </span>
        About Stopwatch
      </a>
      <a
        className="css-1dtgp6n"
        href="#"
      >
        <span>
          <svg
            fill="none"
            height={18}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
            />
            <line
              x1="12"
              x2="12"
              y1="17"
              y2="17"
            />
          </svg>
        </span>
        Help
      </a>
      <a
        className="css-1dtgp6n"
        href="#"
      >
        <span>
          <svg
            fill="none"
            height={18}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="8"
              x2="16"
              y1="15"
              y2="15"
            />
            <line
              x1="9"
              x2="9.01"
              y1="9"
              y2="9"
            />
            <line
              x1="15"
              x2="15.01"
              y1="9"
              y2="9"
            />
          </svg>
        </span>
        Report Issue
      </a>
      <a
        className="css-1dtgp6n"
        href="#"
      >
        <span>
          <svg
            fill="none"
            height={18}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={18}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
        </span>
        Quit Window
      </a>
    </span>
  </div>
</section>
`;

exports[`<Nav /> renders without correctly 1`] = `
<div
  className="css-17qbjpi"
>
  <div
    className="css-147l09t"
  >
    <div
      className="css-jo7s5y"
    />
    <div
      className="css-sz5c6n"
    >
      <div
        className="css-1rmth23"
      >
        <section
          className="css-1vco01a"
        >
          <div
            className="css-k9ylf6"
          >
            <div
              className="css-pfryns"
            >
              <h2>
                Stopwatch
              </h2>
              <span>
                Track time like a thing
              </span>
            </div>
          </div>
          <div
            className="css-1koizjm"
          >
            <span
              className="css-1ago99h"
            >
              <p
                className="css-1pbk3k6"
              >
                Menu
              </p>
              <a
                className="css-1dtgp6n"
                href="#"
              >
                <span>
                  <svg
                    fill="none"
                    height={18}
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    viewBox="0 0 24 24"
                    width={18}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <line
                      x1="12"
                      x2="12"
                      y1="16"
                      y2="12"
                    />
                    <line
                      x1="12"
                      x2="12"
                      y1="8"
                      y2="8"
                    />
                  </svg>
                </span>
                About Stopwatch
              </a>
              <a
                className="css-1dtgp6n"
                href="#"
              >
                <span>
                  <svg
                    fill="none"
                    height={18}
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    viewBox="0 0 24 24"
                    width={18}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <path
                      d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
                    />
                    <line
                      x1="12"
                      x2="12"
                      y1="17"
                      y2="17"
                    />
                  </svg>
                </span>
                Help
              </a>
              <a
                className="css-1dtgp6n"
                href="#"
              >
                <span>
                  <svg
                    fill="none"
                    height={18}
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    viewBox="0 0 24 24"
                    width={18}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <line
                      x1="8"
                      x2="16"
                      y1="15"
                      y2="15"
                    />
                    <line
                      x1="9"
                      x2="9.01"
                      y1="9"
                      y2="9"
                    />
                    <line
                      x1="15"
                      x2="15.01"
                      y1="9"
                      y2="9"
                    />
                  </svg>
                </span>
                Report Issue
              </a>
              <a
                className="css-1dtgp6n"
                href="#"
              >
                <span>
                  <svg
                    fill="none"
                    height={18}
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    viewBox="0 0 24 24"
                    width={18}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                    />
                    <polyline
                      points="16 17 21 12 16 7"
                    />
                    <line
                      x1="21"
                      x2="9"
                      y1="12"
                      y2="12"
                    />
                  </svg>
                </span>
                Quit Window
              </a>
            </span>
          </div>
        </section>
        <span
          className="css-19duvsz"
        >
          <svg>
            logo.svg
          </svg>
        </span>
        <i
          className="css-1f4kraj"
        >
          <svg
            fill="none"
            height={12}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            viewBox="0 0 24 24"
            width={12}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </i>
      </div>
      <div
        className="css-1hvap0d"
      >
        tabs
      </div>
      <div
        className="css-16jk3lx"
      >
        <span
          className="css-10abcsd"
        >
          <span
            className="css-7r5427"
          >
            <span
              className="css-1mgqi2s"
            >
              <svg
                fill="none"
                height={24}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={1.5}
                viewBox="0 0 24 24"
                width={24}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 3v3a2 2 0 0 1-2 2H3m18 0h-3a2 2 0 0 1-2-2V3m0 18v-3a2 2 0 0 1 2-2h3M3 16h3a2 2 0 0 1 2 2v3"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          className="css-10abcsd"
        >
          <span
            className="css-7r5427"
          >
            <span
              className="css-1mgqi2s"
            >
              <svg
                fill="none"
                height={24}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={1.5}
                viewBox="0 0 24 24"
                width={24}
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="5"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="1"
                  y2="3"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="21"
                  y2="23"
                />
                <line
                  x1="4.22"
                  x2="5.64"
                  y1="4.22"
                  y2="5.64"
                />
                <line
                  x1="18.36"
                  x2="19.78"
                  y1="18.36"
                  y2="19.78"
                />
                <line
                  x1="1"
                  x2="3"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="21"
                  x2="23"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="4.22"
                  x2="5.64"
                  y1="19.78"
                  y2="18.36"
                />
                <line
                  x1="18.36"
                  x2="19.78"
                  y1="5.64"
                  y2="4.22"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
